apps/grader-host/src/index.js

```diff
@@ -110,39 +111,50 @@ async.series(
       globalLogger.info('Initialization complete; beginning to process jobs');
       const sqs = new SQSClient(makeAwsClientConfig());
       for (let i = 0; i < config.maxConcurrentJobs; i++) {
-        async.forever((next) => {
-          if (!healthCheck.isHealthy() || processTerminating) return;
-          receiveFromQueue(
-            sqs,
-            config.jobsQueueUrl,
-            (job, done) => {
-              globalLogger.info(`received ${job.jobId} from queue`);
-
-              // Ensure that this job wasn't canceled in the time since job submission.
-              isJobCanceled(job, (err, canceled) => {
-                if (ERR(err, done)) return;
-
-                if (canceled) {
-                  globalLogger.info(`Job ${job.jobId} was canceled; skipping job`);
-                  done();
-                  return;
-                }
-
-                handleJob(job, (err) => {
-                  globalLogger.info(`handleJob(${job.jobId}) completed with err=${err}`);
+        async.forever(
+          (next) => {
+            if (!healthCheck.isHealthy() || processTerminating) return;
+
+            if (!config.jobsQueueUrl) {
+              next(new Error('jobsQueueUrl is not defined'));
+              return;
+            }
+
+            receiveFromQueue(
```

Comment:
I'm looking forward to the eventual async generator version of this!

Reply:
Yes!! This would probably be a really good use case for one of those. We'd have to think carefully about how we'd handle the after-job acknowledgement to the queue though, I'm not sure if async generators can handle that well.

Reply:
Good point, I'm also not quite sure whether there's an elegant way to handle this.